src/main/java/de/tum/in/www1/artemis/service/hestia/TestwiseCoverageService.java

```diff
@@ -295,9 +295,6 @@ public Optional<CoverageReport> getCoverageReportForLatestSolutionSubmissionFrom
      */
     public Optional<CoverageReport> getFullCoverageReportForLatestSolutionSubmissionFromProgrammingExercise(ProgrammingExercise programmingExercise) {
         var optionalLazyReport = getCoverageReportForLatestSolutionSubmissionFromProgrammingExercise(programmingExercise);
-        if (optionalLazyReport.isEmpty()) {
-            return Optional.empty();
-        }
-        return Optional.of(coverageReportRepository.findCoverageReportByIdWithEagerFileReportsAndEntriesElseThrow(optionalLazyReport.get().getId()));
+        return optionalLazyReport.map(coverageReport -> coverageReportRepository.findCoverageReportByIdWithEagerFileReportsAndEntriesElseThrow(coverageReport.getId()));
```

Comment:
We could use `coverageReportRepository::findCoverageReportByIdWithEagerFileReportsAndEntries()` directly because if `optionalLazyReport` is not empty then that report exists and classes that use this method (`BehavioralTestCaseService`, `CoverageReportResource`) have handlers for empty optional anyway.


Btw, this would be more readable with an intermediate `map()`:
```java
optionalLazyReport.map(CoverageReport::getId)
    .map(CoverageReportRepository::findCoverageReportByIdWithEagerFileReportsAndEntries)
```

Reply:
Resolved by adding dedicated database query